### DISCUSSION TITLE
How can I run a single input file using MOOSE?

### DISCUSSION THREAD

**Discussion by WilkAndy** [idaholab/moose#16158] [Q&A General] [answered]
Dear all,

I just begin to learn how to use MOOSE. When I try to run an input file in tutorials, how could I give a command in the terminal? I mean not use ./run_tests for all, but just run one at a time.

Thanks in advance,

Qinglin

**Comment by permcody** [comment]
You use the `-i` flag followed by the name of your input file.

Example:
`./inch-worm-opt -i my_input_file.i`

You can get the complete list of command line options by using the `-h` flag.

**Comment by WilkAndy** [comment, marked as answer]
Since you can `./run_tests` then you must have successfully compiled and tested MOOSE.  Well done!   This is because the compilation will have produced a valid executable that was used by `./run_tests`.  Have a look around in the directory that you used for `./run_tests` and you will probably see an executable called `your_app-opt`, or `combined-opt`, etc.  It is the MOOSE executable.

(If you compiled in debug mode, the executable will have `-dbg` on the end of its name.)

Then, use .... what @permcody just wrote!    You might have to specify the paths, such as
```
./inch-worm-opt -i path/to/my_input_file.i
```
or
```
path/to/inch-worm-opt -i another/path/to/my_input_file.i
```

**Reply by Ali-toghraee** [reply]
Hello Andy,

 I just installed moose (conda) and all the tests passed. I also did the "make" in the combined module folder and created the combined-opt file (however the tests did not run). Now I want to run my input files.
Previously, I used to navigate to the folder of my input file, and then opened that folder in a terminal , then I would typed: 

 (moose) atoghraee@ME-DESKTOP102:~/projects02/ox_poly$ mpiexec -np 88 /home/atoghraee/projects02/moose/modules/combined/combined-opt -i fun05_absM_m8000_df.i

But now , when run this command I face the following errors ( part of that shown below). 
- Is this because the combined-opt tests didnot run?
- I should mention that I already have a moose from 2019 in my projects forlder, to install the new version on moose using conda , I just followed the instructions in the "https://mooseframework.inl.gov/getting_started/installation/conda.html" and instead of "projects" folder , I created an used "projects02" . I would really appreciate if you could help me  make the moose up and running.

(moose) atoghraee@ME-DESKTOP102:~/projects02/ox_poly$ mpiexec -np 88 /home/atoghraee/projects02/moose/modules/combined/combined-opt -i fun05_absM_m8000_df.i
/home/atoghraee/projects02/moose/modules/combined/combined-opt: /opt/moose/gcc-7.3.0/lib64/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /home/atoghraee/projects02/moose/modules/combined/test/lib/libcombined_test-opt.so.0)
/home/atoghraee/projects02/moose/modules/combined/combined-opt: /opt/moose/gcc-7.3.0/lib64/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /home/atoghraee/projects02/moose/modules/combined/lib/libcombined-opt.so.0)
/home/atoghraee/projects02/moose/modules/combined/combined-opt: /opt/moose/gcc-7.3.0/lib64/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /home/atoghraee/projects02/moose/framework/libmoose-opt.so.0)
/home/atoghraee/projects02/moose/modules/combined/combined-opt: /opt/moose/gcc-7.3.0/lib64/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /home/atoghraee/projects02/moose/modules/chemical_reactions/test/lib/libchemical_reactions_test-opt.so.0)
/home/atoghraee/projects02/moose/modules/combined/combined-opt: /opt/moose/gcc-7.3.0/lib64/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /home/atoghraee/projects02/moose/modules/fluid_properties/test/lib/libfluid_properties_test-opt.so.0)
/home/atoghraee/projects02/moose/modules/combined/combined-opt: /opt/moose/gcc-7.3.0/lib64/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /home/atoghraee/projects02/moose/modules/ray_tracing/test/lib/libray_tracing_test-opt.so.0)
/home/atoghraee/projects02/moose/modules/combined/combined-opt: /opt/moose/gcc-7.3.0/lib64/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /home/atoghraee/projects02/moose/modules/heat_conduction/test/lib/libheat_conduction_test-opt.so.0)
/home/atoghraee/projects02/moose/modules/combined/combined-opt: /opt/moose/gcc-7.3.0/lib64/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /home/atoghraee/projects02/moose/modules/level_set/test/lib/liblevel_set_test-opt.so.0)
/home/atoghraee/projects02/moose/modules/combined/combined-opt: /opt/moose/gcc-7.3.0/lib64/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /home/atoghraee/projects02/moose/modules/misc/test/lib/libmisc_test-opt.so.0)
/home/atoghraee/projects02/moose/modules/combined/combined-opt: /opt/moose/gcc-7.3.0/lib64/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /home/atoghraee/projects02/moose/modules/rdg/test/lib/librdg_test-opt.so.0)
/home/atoghraee/projects02/moose/modules/combined/combined-opt: /opt/moose/gcc-7.3.0/lib64/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /home/atoghraee/projects02/moose/modules/navier_stokes/test/lib/libnavier_stokes_test-opt.so.0)
/home/atoghraee/projects02/moose/modules/combined/combined-opt: /opt/moose/gcc-7.3.0/lib64/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /home/atoghraee/projects02/moose/modules/tensor_mechanics/test/lib/libtensor_mechanics_test-opt.so.0)
/home/atoghraee/projects02/moose/modules/combined/combined-opt: /opt/moose/gcc-7.3.0/lib64/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by
